### PR TITLE
Master wowl tooltip aab

### PR DESCRIPTION
**Pull request by aab-odoo**

### CODE REVIEW

**Comment by aab-odoo** [comment]
robodoo rebase-ff

File: addons/web/static/src/core/listener_hook.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    if (typeof arguments[1] !== "string") {
+        querySelector = null;
+        handler = arguments[1];
+        addEventListenerOptions = arguments[2];
+    }
+    if (typeof handler !== "function") {
+        throw new Error("The handler must be a function");
+    }
+
+    const comp = Component.current;
```
Comment (brboi) [inline addons/web/static/src/core/listener_hook.js]: prefer useComponent ?

File: addons/web/static/src/core/listener_hook.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ * Note: components that alter the event's target (e.g. Portal) are not
+ * expected to behave as expected with event delegation.
+ *
+ * @param {string} eventName the name of the event
+ * @param {string} [querySelector] a JS native selector for event delegation
+ * @param {function} handler the event handler (will be bound to the component)
+ * @param {Object} [addEventListenerOptions] to be passed to addEventListener as
+ *   options. Useful for listening in the capture phase
+ */
+export function useListener(eventName, querySelector, handler, addEventListenerOptions) {
```
Comment (brboi) [inline addons/web/static/src/core/listener_hook.js]: should be tested.
e.g. with { capture: true } in conjunction with selector

File: addons/web/static/src/core/tooltip/tooltip_hook.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * The ideal position of the tooltip can be specified thanks to the attribute
+ * "data-tooltip-position":
+ *   <button data-tooltip="This is a tooltip" data-tooltip-position="left">Do something</button>
+ */
+export function useTooltip() {
+    const popover = usePopover();
+    let tooltipTimeout;
+    let closeTooltip;
+    function cleanup() {
+        clearTimeout(tooltipTimeout);
```
Comment (brboi) [inline addons/web/static/src/core/tooltip/tooltip_hook.js]: prefer browser.clearTimeout

File: addons/web/static/src/core/tooltip/tooltip_hook.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            return cleanup(); // target is no longer in the DOM
+        }
+        const targetRect = target.getBoundingClientRect();
+        if (
+            positionX < targetRect.left ||
+            positionX > targetRect.right ||
+            positionY < targetRect.top ||
+            positionY > targetRect.bottom
+        ) {
+            return cleanup(); // mouse is no longer hovering the target
```
Comment (kebeclibre) [inline addons/web/static/src/core/tooltip/tooltip_hook.js:83]: I think we also want to keep the tooltip if we are hovering it (for copy/paste purpose when the tooltip has such info).

**Comment by kebeclibre** [comment]
robodoo r+

**Pull request closed without merging**